storage/tree_validation.go

```diff
@@ -64,6 +64,21 @@ func ValidateTreeForCreation(ctx context.Context, tree *trillian.Tree) error {
 	return validateMutableTreeFields(ctx, tree)
 }
 
+func validateTreeTypeUpdate(oldTree, newTree *trillian.Tree) error {
+	const prefix = "changing tree_type"
```

Comment:
I still find it hard to look at this code and decide if it's doing the right thing. I think the documentation of what it's validating should be included here.

Reply:
Done.